crates/solver/src/main.rs

```diff
@@ -447,6 +456,20 @@ async fn main() {
     };
     tokio::task::spawn(maintainer.run_maintenance_on_new_block(current_block_stream));
 
+    let periodic_update_token_list = async move {
+        loop {
```

Comment:
I feel this part should live inside the token_list module and also that we should have a type defined for a shared and updatable token list. Something like

```rust
/// Token list that updates periodically from a URL and can be shared by cloning.
#[derive(Clone)]
pub struct UpdatingTokenList {
    inner: Arc<Mutex<TokenList>>,
}

impl UpdatingTokenList {
    // Perform the initial update and spawn a background task that updates forever
    pub async fn new(config) -> Result<Self> {
        // background task takes ownership of client, url, update interval
}

    // neither async nor returns result
    pub fn get(&self) -> TokenList {}
}
```

This way we don't expose any of the internals like what type is used to accomplish the sharing and how the updating works. Components that need the list just get a clone of the UpdatingTokenList we create once in main.

Reply:
Sorry for only writing about this now. It didn't occur to me before to do it like this.

Reply:
@vkgnosis Kudos, this is a really good idea. The only thing I dislike is this `UpdatingTokenList` abstraction. Again my domain knowledge is lacking, but it seems like the `TokenList` is always used behind an `Arc<RwLock<>>` to begin with. I think it'd be better to move the `Arc<RwLock<>>` into `TokenList` and then either start the updater task in `new` or something similar. What do you think?

Reply:
Depends on if we also want to use TokenList in places where it wouldn't be behind Arc and Mutex. I think you're right that it currently isn't the case so your suggestion works too.

Reply:
Moved `Arc<RwLock<>>` to `TokenList`, also moved updater to `TokenList` constructor.
Ended up with a generally much cleaner code.